risc0/zkvm/r0vm/src/main.rs

```diff
@@ -0,0 +1,138 @@
+// Copyright 2022 Risc0, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use clap::Parser;
+use std::fs;
+use std::io::Write;
+
+use risc0_zkvm_host::{MethodId, Prover, Receipt, DEFAULT_METHOD_ID_LIMIT};
+
+/// Generates a MethodID for a given RISC-V ELF binary.
+#[derive(Parser)]
+#[clap(about, version, author)]
+struct Args {
+    /// The ELF file to run
+    #[clap(long)]
+    elf: String,
+
+    /// MethodID file; created if it doesn't exist.
+    #[clap(long)]
+    method_id: Option<String>,
+
+    /// Proof output file.
+    #[clap(long)]
+    proof: Option<String>,
+
+    /// File to read initial input from.
+    #[clap(long)]
+    initial_input: Option<String>,
+
+    /// Display verbose output.
+    #[clap(short, long, action = clap::ArgAction::Count)]
+    verbose: u8,
+
+    /// Limit the number of hash table entries to compute.
```

Comment:
To clarify, it's the number of hash entries in the MethodID table (might be confusing to people if they associate hash table a standard CS hashtable).